File: Algorithm-Toolbox/week1/MaximumPairwiseProduct.java
import java.util.Scanner;

class MaximumPairwiseProduct {

  public static void main(String[] args) {
    Scanner inputUser = new Scanner(System.in);
    Integer sizeArray = inputUser.nextInt();
    long[] elements = new long[sizeArray];

    for (int i = 0; i < sizeArray; ++i) {
      elements[i] = inputUser.nextInt();
    }

    int positionMaximumValue = 0;
    int positionSecondMaximumValue = -1;

    for (int i = 0; i < sizeArray; ++i) {
      if (elements[i] > elements[positionMaximumValue]) {
        positionMaximumValue = i;
      }
    }

    for (int i = 0; i < sizeArray; ++i) {
      if ( i != positionMaximumValue) {
        if (positionSecondMaximumValue == -1 ||  elements[i] > elements[positionSecondMaximumValue]){
         positionSecondMaximumValue = i;
        
        }
      }
    }

    System.out.println(elements[positionMaximumValue] * elements[positionSecondMaximumValue]);
  }
}

File: Algorithm-Toolbox/week1/maximumPairwiseProduct.cpp
#include <iostream>

int main() {
    int sizeArray;
    std::cin >> sizeArray;    
    long long elements[sizeArray];
    long long maxPairwise = 0;
    long long productActual = 0;

    for(int i=0; i< sizeArray; ++i) {
        std::cin >> elements[i];
    }
    
    for(int i=0; i< sizeArray; ++i) {
        for(int j=0; j< sizeArray; ++j) {
            productActual = elements[i]*elements[j];
            if ( productActual > maxPairwise && i != j)
                maxPairwise = productActual;
        }
    }

    std::cout << maxPairwise;

    return 0;
}